### PR TITLE
feature/TIN-15

### PR DESCRIPTION
**Pull request by tsenovi**

### CODE REVIEW

File: src/main/java/com/volasoftware/tinder/models/Account.java [hunk 1]
```diff
@@ -1,18 +1,23 @@
...
+import jakarta.persistence.FetchType;
 import jakarta.persistence.GeneratedValue;
 import jakarta.persistence.GenerationType;
 import jakarta.persistence.Id;
 
 import jakarta.persistence.JoinColumn;
 import jakarta.persistence.JoinTable;
 import jakarta.persistence.ManyToMany;
+import jakarta.persistence.OneToOne;
+import jakarta.persistence.PrimaryKeyJoinColumn;
```
Comment (todornikolov1) [inline src/main/java/com/volasoftware/tinder/models/Account.java]: Unused?

File: src/main/java/com/volasoftware/tinder/services/contracts/FriendService.java [hunk 2]
```diff
@@ -1,8 +1,15 @@
 package com.volasoftware.tinder.services.contracts;
 
+import java.util.Optional;
```
Comment (todornikolov1) [inline src/main/java/com/volasoftware/tinder/services/contracts/FriendService.java]: Any use of it?

File: src/main/java/com/volasoftware/tinder/models/Location.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.MapsId;
+import jakarta.persistence.OneToOne;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
```
Comment (todornikolov1) [inline src/main/java/com/volasoftware/tinder/models/Location.java]: All used?

File: src/main/java/com/volasoftware/tinder/services/implementations/FriendServiceImpl.java [hunk 4]
```diff
@@ -50,6 +62,85 @@ public void removeFriend(Long friendId) {
...
+        if (!hasFriends(realAccount)) {
+            realAccount.setFriends(new HashSet<>());
+        }
+
+        Account savedAccount = addBotFriendsToExistingSet(realAccount, botSet);
+
+        int finalFriendsCount = getFriendsCount(savedAccount);
+        if (finalFriendsCount > initialFriendsCount) {
+            return OperationConstant.SUCCESSFUL;
+        } else {
```
Comment (todornikolov1) [inline src/main/java/com/volasoftware/tinder/services/implementations/FriendServiceImpl.java]: no need of else

File: src/main/java/com/volasoftware/tinder/services/implementations/FriendServiceImpl.java [hunk 5]
```diff
@@ -50,6 +62,85 @@ public void removeFriend(Long friendId) {
...
+    }
+
+    private int getFriendsCount(Account account) {
+        return account.getFriends() == null ? 0 : account.getFriends().size();
+    }
+
+    private int getFriendsCount(Set<Account> realAccounts) {
+        return realAccounts
+            .stream()
+            .mapToInt(account -> account.getFriends() == null ? 0 : account.getFriends().size())
```
Comment (todornikolov1) [inline src/main/java/com/volasoftware/tinder/services/implementations/FriendServiceImpl.java]: DRY principle. Just call getFriendsCount(account)

File: src/main/resources/db/migration/V11.01__undo_adding_records_to_accounts.sql [hunk 6]
```diff
@@ -0,0 +1,2 @@
+DELETE FROM account WHERE email = 'ivan.t.tsenov@gmail.com';
```
Comment (todornikolov1) [inline src/main/resources/db/migration/V11.01__undo_adding_records_to_accounts.sql]: No need of new migration file, just put these deletes in a single migration fail with other code

File: src/main/resources/db/migration/V11.02__add_initial_accounts_with_encrypted_pass.sql [hunk 7]
```diff
@@ -0,0 +1,7 @@
+SET @current_date = NOW();
```
Comment (todornikolov1) [inline src/main/resources/db/migration/V11.02__add_initial_accounts_with_encrypted_pass.sql]: No need of new migration file, just put all code from the task in one migration file

File: src/main/java/com/volasoftware/tinder/models/Account.java (hunk 1 above)
Comment (tsenovi) [reply src/main/java/com/volasoftware/tinder/models/Account.java]: Unused and removed.

File: src/main/java/com/volasoftware/tinder/services/contracts/FriendService.java (hunk 2 above)
Comment (tsenovi) [reply src/main/java/com/volasoftware/tinder/services/contracts/FriendService.java]: Unused and removed.

File: src/main/java/com/volasoftware/tinder/models/Location.java (hunk 3 above)
Comment (tsenovi) [reply src/main/java/com/volasoftware/tinder/models/Location.java]: Yes, except for the MapsId import, which will be removed. The rest will be used in the next tasks, I believe.

File: src/main/java/com/volasoftware/tinder/services/implementations/FriendServiceImpl.java (hunk 4 above)
Comment (tsenovi) [reply src/main/java/com/volasoftware/tinder/services/implementations/FriendServiceImpl.java]: I agree. "else" will be removed.

**Pull request merged**